### PR TITLE
chore: cleanup and sync with bentoml

### PR DESCRIPTION
**Pull request by aarnphm**
- remove docker python dependencies in light for using docker binary
- in sync with requirements for pyyaml with bentoml upstream

Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

### CODE REVIEW

File: bentoctl/cli/__init__.py [hunk 1]
```diff
@@ -142,36 +142,26 @@ def build(
...
+        repository_url, _, _ = deployment_config.create_repository()
         console.print(f"Created the repository {deployment_config.repository_name}")
-        repository_image_tag = deployment_config.generate_docker_image_tag(
-            repository_url
-        )
-        tag_docker_image(local_docker_tag, repository_image_tag)
-        push_docker_image_to_repository(
-            repository=repository_image_tag,
-            username=username,
-            password=password,
```
Comment (yubozhao) [inline bentoctl/cli/__init__.py:168]: User can push to ECR with username/password, without logging into with docker CLI locally. With the new implementation, does the user have to login first?

Comment (aarnphm) [reply bentoctl/cli/__init__.py:168]: oh right, let me add that

File: bentoctl/docker_utils.py [hunk 2]
```diff
@@ -94,49 +72,27 @@ def generate_deployable_container(
...
+    disable_content_trust: bool | None = None,
+    *,
+    username: str | None = None,
+    password: str | None = None,
+) -> None:
+    if username:
+        assert password, "password is required when username is provided"
+        login_cmd = ["docker", "login"]
+        login_cmd.extend(["-u", username, "-p", password])
+        run_subprocess_cmd(login_cmd)
```
Comment (yubozhao) [inline bentoctl/docker_utils.py:90]: Is there a way we pass in username and password only when we are pushing the image?

I don't want to override user's current logged in account on docker CLI

**Comment by yubozhao** [comment]
Let's merge this PR after we have some Github Action test

**Pull request closed without merging**